include/lsst/afw/geom/Functor.h

```diff
@@ -0,0 +1,99 @@
+// -*- lsst-c++ -*-
+
+/* 
+ * LSST Data Management System
+ * Copyright 2015 LSST Corporation.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+/**
+ * \file 
+ * @brief Functor classes used by SeparableXYTransform. 
+ */
+
+#ifndef LSST_AFW_GEOM_FUNCTOR_H
+#define LSST_AFW_GEOM_FUNCTOR_H
+
+#include <string>
+#include "boost/shared_ptr.hpp"
+#include "lsst/base.h"
+#include "lsst/daf/base.h"
+
+namespace lsst {
+namespace afw {
+namespace geom {
+
+/// @brief Abstract base class for function objects.
+class Functor : public daf::base::Citizen {
+
+public:
+
+   /// @param owner The name of the class that owns a concrete
+   ///              instance of this class.  Used in the exception
+   ///              message from the inverse(...) member function.
+   explicit Functor(std::string const & owner);
+
+   virtual ~Functor() {}
+
+   virtual PTR(Functor) clone() const = 0;
+
+   /// @return y = f(x)
+   virtual double operator()(double x) const = 0;
+
+   /// @return df(x)/dx evaluated at x.  This is used in the
+   /// inverse(...) member function.
+   virtual double deriv(double x) const = 0;
```

Comment:
Should this be a public method (is it likely to be useful by others) or can it be protected (in which case it should have a leading underscore in the name).

In any case I suggest expanding the name to derivative, since our naming standard frowns on abbreviations.